types/chartmogul-node/index.d.ts

```diff
@@ -14,23 +14,50 @@ export namespace Ping {
 }
 
 export namespace DataSource {
+    type InvoiceHandlingTrigger = "open" | "paid" | string | undefined;
+    type DataSourceStatus = "never_imported" | "import_complete" | "import_in_progress" | "import_failed" | string | undefined;
+
+    interface ProcessingStatus {
+        processed?: number;
+        pending?: number;
+        failed?: number;
+    }
+    interface InvoiceHandlingMode {
+        create_subscription_when_invoice_is: InvoiceHandlingTrigger;
+        update_subscription_when_invoice_is: InvoiceHandlingTrigger;
+        prevent_subscription_for_invoice_voided: boolean;
+        prevent_subscription_for_invoice_refunded: boolean;
+        prevent_subscription_for_invoice_written_off: boolean;
+    }
+    interface InvoiceHandlingSetting {
+        manual?: InvoiceHandlingMode;
+        automatic?: InvoiceHandlingMode;
+    }
+    interface ExtraDataSourceParams {
+        with_processing_status?: boolean;
+        with_auto_churn_subscription_setting?: boolean;
+        with_invoice_handling_setting?: boolean;
+    }
     interface DataSource {
-        uuid?: string | undefined;
```

Comment:
A lot of `| undefined` in here, did you figure out a reason for why almost everything was defined like this?